File: back/TalkyAPI/Models/Story.cs
namespace TalkyAPI.Models
{
    public class Story
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public string? Caption { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime ExpiresAt { get; set; } = DateTime.UtcNow.AddHours(24);
        
        // Navigation
        public User User { get; set; } = null!;
        public ICollection<StoryView> Views { get; set; } = new List<StoryView>();
    }

    public class StoryView
    {
        public int Id { get; set; }
        public int StoryId { get; set; }
        public int UserId { get; set; }
        public DateTime ViewedAt { get; set; } = DateTime.UtcNow;
        
        // Navigation
        public Story Story { get; set; } = null!;
        public User User { get; set; } = null!;
    }
}

File: src/types/story.ts
export interface Story {
  id: number;
  userId: number;
  username: string;
  imageUrl: string;
  caption?: string;
  createdAt: string;
  expiresAt: string;
  viewCount: number;
  hasViewed: boolean;
}

export interface CreateStoryDto {
  imageUrl: string;
  caption?: string;
}

export interface StoryView {
  id: number;
  storyId: number;
  userId: number;
  username: string;
  viewedAt: string;
}
